### DISCUSSION TITLE
How do i query multiple records based on a one to many relationship

### DISCUSSION THREAD

**Discussion by busyLambda** [pocketbase/pocketbase#3400] [Q&A]
So let's say we have a user with id 1
There is a collection of tasks with an assigned_to field that is a reference to many users.
I wish to get all tasks where user with id 1 is present as a user assigned to do the task.
There doesn't seem to be anything on this in the docs as far as i've looked.

**Comment by ganigeorgiev** [comment]
You can use dot-notation and the "any/at-least-one" operators. Here is an example with the JS SDK:

```js
const userId = "..."

const tasks = await pb.collection("tasks").getFullList({
    filter: `assigned_to.id ?= "${userId}"`
})
```

**Reply by ganigeorgiev** [reply]
Depending on your use case, you can also set the following List API rule for your `tasks` collection (_assuming `assigned_to` is marked as nonempty_):

```js
assigned_to.id ?= @request.auth.id
```

Which will automatically filter and allow access to only the tasks that the authenticated user is assigned to and on the client-side you can just use:

```js
await pb.collection("users").authWithPassword("test@exaple.com", "1234567890")

...

const tasks = await pb.collection("tasks").getFullList()
```
